README.md

```diff
@@ -1,85 +1,711 @@
+| pristine | Returns false when the form gets spoilt. <br> helps to keep track with the user's interaction with the form. |
+| error    | The error state of form.                                                                                     |
+
+  ```typescript
+  type FormInput = {
+    userName?: string;
+  };
+
+  const {  } = useForm<FormInput>(
+    schema,
+    {
+      value: {},
+      pristine: initialPristine,
+      error: initialError,
+    },
+  );
+  ```
+
+**The return value from useForm hooks are as follows:**
+  | value    | description                             |
+  | -------- | --------------------------------------- |
+  | value    | The value of the form                   |
+  | error    | The error state of the form             |
+  | pristine | Defines the pristine state of the form. |
+
+- ### setFieldValue
+    setFieldValue is used to set the value of each input individually in a form.
+
+  ``` typescript
+  //setFieldValue('value', 'fieldName');
+   setFieldValue('Test name', 'name' as const);
+    <RawInput
+      name='name'
+      value={value.name
+      onChange={setFieldValue}
+    />
+  ```
+- ### setValue
+  Each field is required to have a name as a key to set value process. <br>
+  This is used to set the value for a group of input as it also   returns the old value. <br>
+  Generally, the use case will be to mapped the data fetched from API according to the concern field. <br>
+
+  ``` typescript
+  const data = React.useMemo(() => ({
+    name: 'test name',
+    age: '25',
+  }), []);
+
+  React.useEffect(() => {
+    setValue(data);
+  }, [data, setValue]);
+
+  // OR,
+  //React.useEffect(() => {
+  //  setValue({
+  //    name: data.name,
+  //  });
+  //}, [data]);
+
+  return (
+    <RawInput
+      name='name'
+      value={value.name}
+      onChange={setFieldValue}
+    />
+    <RawInput
+      name='age'
+      value={value.age}
+      onChange={setFieldValue}
+    />
+  )
+
+  ```
+
+- ### setPristine
+  Pristine is used to decide whether to disable form submission or not. Whenever any event occurs in the form it gets enabled and vice-versa.
+
+  ``` typescript
+    <form>
+      <RawInput
+        name='username'
+        value={value.userName}
+        onChange={setFieldValue}
+      />
+      <Button disabled={pristine}>
+        submit
+      </Button>
+    </form>
+  ```
+  </details>
+
+- ### setError
+  This function is used to set one or more errors manually.
+  Errors will return when validations fails.
+
+### getObjectError
+type conversion of errors as object.
+
+```typescript
+  const {
+    error: riskyError,
+  } = useForm(
+    schema,
+    {
+      value:{},
+    },
+    );
+
+  const error = getErrorObject(riskyError);
+```
 
+### createSubmitHandler
+  `createSubmitHandler`is invoked on form submission. This function will receive the form data if the validation is successful. All the utility functions passed as parameters are used sequentially.
```

Comment:
What do you mean by "All the utility functions passed as parameters are used sequentially"?